pkg/controllers/jobset_controller.go

```diff
@@ -129,6 +130,59 @@ func (r *JobSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, nil
 	}
 
+	// If JobSpec has a suspend field of true,
+	// set condition to say jobspec is suspended.
+	jobsetSuspended := js.Spec.Suspend != nil && *js.Spec.Suspend
+	if jobsetSuspended {
+		if err := r.ensureCondition(ctx, &js, corev1.EventTypeNormal, metav1.Condition{
+			Type:               string(jobset.JobSetSuspended),
+			Status:             metav1.ConditionStatus(corev1.ConditionTrue),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "SuspendedJobs",
+			Message:            "jobset is suspended",
+		}); err != nil {
+			log.Error(err, "updating jobset status")
+			return ctrl.Result{}, nil
+		}
+		// If jobs are started and jobset suspends, we need to update jobs
+		// to be suspended.
+		for _, job := range ownedJobs.active {
+			if !pointer.BoolDeref(job.Spec.Suspend, false) {
+				job.Spec.Suspend = pointer.Bool(true)
+				if r.Update(ctx, job); err != nil {
+					log.Error(err, "updating job suspend")
+					return ctrl.Result{}, nil
+				}
+			}
+		}
+
+		// If JobSpec is unsuspended either by removing suspend field or setting to false
+		// and we detect a condition that says it was suspended
+		// we must update all jobs to be suspended and modify condition.
+	} else {
+		if jobSetSuspended(&js) {
```

Comment:
ensureCondition does a few things:

1) Replaces a condition if it exists but ConditionStatus is different.
2) Appends if condition doesn't exist
3) if duplicate then does nothing

So either 1 or 2 will then force an update.  So I think we still need this check because this guards against the case that if a job was previously suspended, then we know that we should resume.  Otherwise we don't need to set that condition.

Reply:
But maybe we should change the name then.  

This was UpdateStatusWithCondition but we changed it to only update in case of condition being changed or non existent.